### PR TITLE
YONK-791: Mobile app Detail & List APIs with Unit Tests

### PR DESCRIPTION
**Pull request by aamishbaloch**
These changes are done to integrate **`mobileapps-edx-platform-extensions`** with API integration and Platform. Changes include addition of mobile app url, with some new permissions used in mobile apps and an extension to Username Cipher for encryption of secret keys used for mobile apps.

UPDATE: **`SecureAPIView`** is now inherited by **`PermissionMixin`** to enforce more security to our views.

UPDATE: Documentation of mobile app management APIs added.

### CODE REVIEW

File: edx_solutions_api_integration/permissions.py [hunk 1]
```diff
@@ -219,6 +219,36 @@ class SecureListAPIView(PermissionMixin,
...
+                        PaginationMixin,
+                        generics.ListCreateAPIView):
+    """
+    Inherited from ListCreateAPIView
+    """
+    pass
+
+
+class SecureRetrieveUpdateDestroyAPIView(PermissionMixin,
+                        FilterBackendMixin,
```
Comment (ziafazal) [inline edx_solutions_api_integration/permissions.py]: I think we don't need `FilterBackendMixin` and `PaginationMixin` for this view.

File: edx_solutions_api_integration/permissions.py [hunk 2]
```diff
@@ -219,6 +219,36 @@ class SecureListAPIView(PermissionMixin,
...
+                        PaginationMixin,
+                        generics.RetrieveUpdateDestroyAPIView):
+    """
+    Inherited from RetrieveUpdateDestroyAPIView
+    """
+    pass
+
+
+class SecureRetrieveUpdateAPIView(PermissionMixin,
```
Comment (ziafazal) [inline edx_solutions_api_integration/permissions.py]: I think we don't need `FilterBackendMixin` and `PaginationMixin` for this view.

File: edx_solutions_api_integration/test_utils.py [hunk 3]
```diff
@@ -192,6 +191,16 @@ def do_put(self, uri, data):
         return self.client.put(
             uri, headers=headers, content_type='application/json', data=json_data)
 
+    def do_patch(self, uri, data):
+        """Submit an HTTP PUT request"""
```
Comment (ziafazal) [inline edx_solutions_api_integration/test_utils.py]: typo: HTTP PUT should be HTTP PATCH

File: edx_solutions_api_integration/utils.py [hunk 4]
```diff
@@ -411,3 +411,14 @@ def get_profile_image_urls_by_username(username, profile_image_uploaded_at):
         for size_display_name, url in urls.items()
     })
     return data
+
+
+class StringCipher(UsernameCipher):
```
Comment (ziafazal) [inline edx_solutions_api_integration/utils.py]: why not simple write an empty class which inherits from `UsernameCipher `?
```
class StringCipher(UsernameCipher):
    pass
```

**Comment by aamishbaloch** [comment]
@ziafazal done with the changes, Can you review?

**Comment by aamishbaloch** [comment]
UPDATE: **`SecureAPIView`** is now inherited by **`PermissionMixin`** to enforce more security to our views.

**Comment by aamishbaloch** [comment]
UPDATE: Documentation of mobile app management APIs added.

**Pull request merged**